### DISCUSSION TITLE
Hero 9 USB power trigger

### DISCUSSION THREAD

**Discussion by Espusen** [gopro/labs#672] [Q&A]
Hi all

I am trying to get my HERO 9 to switch on when USB power is enabled, take one image (Night photo), then switch off.
I have used the following code:
!MBOOT="!Lbt"!SAVEbt=mPN>u0!S<u0!E+!1O!R8

It switches on, then immediately switches off again.
Blockly diagram below. Any ideas what I am doing wrong?
<img width="1430" alt="Screenshot 2023-12-09 at 22 48 58" src="https://github.com/gopro/labs/assets/153327756/cf7b7825-addc-4d06-80fe-fde33b1f8427">

**Comment by dnewman-gpsw** [comment]
The code Blockly generates used commands that are only available on HERO11/12/MAX(limited) and Mini.  The USB power test as a conditional was introduced after the HERO9/10 Labs releases.

**Comment by Espusen** [comment]
Ok I understand
Many thanks for shedding light on this.
If The HERO 12 really is more reliable for the LABS scripts I might have to invest in one!

Is there a blockly tutorial anywhere??
